File: frontEnd/src/types/dto/stepDTO/AssertStepDTO.ts
// 断言操作
import {AssertType} from "@/types/enum/condation/AssertType";
import {AssertRelationship} from "@/types/enum/condation/AssertRelationship";
import {PageAttribute} from "@/types/enum/page/PageAttribute";

import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";
import {ElementDTO} from "@/types/dto/ElementDTO";

export class AssertStepDTO extends StepBaseDTO {
    stepType: string = 'ASSERT';
    assertType: AssertType | string = 'ELEMENT_EXIST' // 断言类型
    element?: ElementDTO = new ElementDTO(); // 断言元素
    assertText?: string = ''; // 断言文本
    elementAttribute?: string = "value"; // 断言元素属性
    assertRelationship?: AssertRelationship | string = "EQUALS"; // 断言关系
    exceptValue?: string; // 期望值
    pageAttribute?: PageAttribute | string = "PAGE_TITLE"; // 断言元素属性
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/AssertStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;


import com.mokatest.platform.demos.domain.ui.uiEnum.condation.AssertRelationship;
import com.mokatest.platform.demos.domain.ui.uiEnum.page.PageAttribute;
import com.mokatest.platform.demos.domain.ui.dto.ElementDTO;
import lombok.Data;

// 断言操作
@Data
public class AssertStepDTO extends StepBaseDTO {
    private String stepType;
    private String assertType; // 断言类型
    private ElementDTO element; // 断言元素
    private String assertText; // 断言文本
    private String elementAttribute; // 断言元素属性
    private AssertRelationship assertRelationship; // 断言关系
    private String exceptValue; // 期望值
    private PageAttribute pageAttribute; // 断言元素属性
}






